core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1520,6 +1534,30 @@ class SparkContext(config: SparkConf) extends Logging {
    */
   def listFiles(): Seq[String] = addedFiles.keySet.toSeq
 
+  /**
+   * Add an archive to be downloaded and unpacked with this Spark job on every node.
+   *
+   * If an archive is added during execution, it will not be available until the next TaskSet
```

Comment:
Is this limitation applied to the other ways like `spark.files` too?

Reply:
Yeah, it does. It was actually copied from `addFile`.